### ISSUE TITLE
support for URLSearchParams

### ISSUE THREAD

**Issue by pelikhan** [microsoft/devicescript#506]
Implement URLSearchParams

- mdn: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams
- guide : https://microsoft.github.io/devicescript/contributing#add-core-javascript-apis

**Comment by arnabrahman** [comment]
I would like to take a crack at this.

**Comment by arnabrahman** [comment]
For prerequisites of this, we need the `array.sort()` & `decodeURIComponent()` function implementations. what do you think? @pelikhan

**Comment by pelikhan** [comment]
Yes let's add them! I suggest preparing individual PRs for each of them.

**Comment by pelikhan** [comment]
Added https://github.com/microsoft/devicescript/issues/599

**Comment by pelikhan** [comment]
Added https://github.com/microsoft/devicescript/issues/600
